src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,387 @@
+import React from 'react'
+import {Box} from '..'
+import {isResponsiveValue, useResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {SxProp, merge, BetterSystemStyleObject} from '../sx'
+import Heading from '../Heading'
+import {ArrowLeftIcon} from '@primer/octicons-react'
+import Link from '../Link'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+import {get} from '../constants'
+const REGION_ORDER = {
+  ContextArea: 0,
+  TitleArea: 1,
+  Description: 2,
+  Navigation: 3,
+}
+
+// Types that are shared between sub components
+export type sharedPropTypes = {
+  hidden?: boolean | ResponsiveValue<boolean>
+} & SxProp
+
+// Default state for the `visible` prop when a sub component is only visible on narrow viewport
+const hiddenOnRegularAndWide = {
+  narrow: false,
+  regular: true,
+  wide: true,
+}
+
+// Default state for the `visible` prop when a sub component is visible on regular and wide viewport
+const hiddenOnNarrow = {
+  narrow: true,
+  regular: false,
+  wide: false,
+}
+
+// Root
+// -----------------------------------------------------------------------------
+export type PageHeaderProps = {
+  'aria-label'?: React.AriaAttributes['aria-label']
+  // TODO: check if we need any other types
+  as?: React.ElementType | 'header' | 'div'
+} & sharedPropTypes
+
+const Root: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, as = 'div'}) => {
+  const rootStyles = {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: get('space.2'),
+    // 24px for wide and regular, 16px for narrow viewports
+    padding: useResponsiveValue(
+      {
+        narrow: 3,
+        regular: 4,
+        wide: 4,
+      },
+      false,
+    ),
+  }
+  return (
+    <Box as={as} sx={merge<BetterSystemStyleObject>(rootStyles, sx)}>
+      {children}
+    </Box>
+  )
+}
+
+function displayResponsively<T, F>(value: T, fallback: F): BetterSystemStyleObject {
```

Comment:
Ahh, sorry I forgot to mention about it. It is an experiment to use CSS breakpoints rather than relying on using `useResponsiveHook` and it comes from this [convo] and I only used it for one sub component. (https://github.com/primer/react/pull/2265#discussion_r1036355747) I am actually thinking to tackle this in a different PR, this one is already big enough 😆

Reply:
If we're going to tackle this in a separate PR, do you think we should create an issue for that work once this merges?

Reply:
Agreed! https://github.com/github/primer/issues/1569 done!